common/src/main/java/com/revenuecat/purchases/common/verification/SignatureVerificationMode.kt

```diff
@@ -6,21 +6,29 @@ sealed class SignatureVerificationMode {
     companion object {
         fun fromEntitlementVerificationMode(
             verificationMode: EntitlementVerificationMode,
-            signatureVerifier: SignatureVerifier? = null,
+            rootVerifier: SignatureVerifier? = null,
         ): SignatureVerificationMode {
             return when (verificationMode) {
                 EntitlementVerificationMode.DISABLED -> Disabled
                 EntitlementVerificationMode.INFORMATIONAL ->
-                    Informational(signatureVerifier ?: DefaultSignatureVerifier())
+                    Informational(IntermediateSignatureHelper(rootVerifier ?: DefaultSignatureVerifier()))
                 // Hidden ENFORCED mode during feature beta
                 // EntitlementVerificationMode.ENFORCED ->
                 //     Enforced(signatureVerifier ?: DefaultSignatureVerifier())
             }
         }
     }
     object Disabled : SignatureVerificationMode()
-    data class Informational(val signatureVerifier: SignatureVerifier) : SignatureVerificationMode()
-    data class Enforced(val signatureVerifier: SignatureVerifier) : SignatureVerificationMode()
+    data class Informational(
+        override val intermediateSignatureHelper: IntermediateSignatureHelper = IntermediateSignatureHelper(
+            DefaultSignatureVerifier(),
+        ),
+    ) : SignatureVerificationMode()
+    data class Enforced(
+        override val intermediateSignatureHelper: IntermediateSignatureHelper = IntermediateSignatureHelper(
+            DefaultSignatureVerifier(),
+        ),
```

Comment:
Nit: can you now extract this to something like `createSignatureHelper()`?

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1111/commits/291fc74254b359c9c3b134d46024aa93b0f2ff16